lib/index.js

```diff
@@ -106,24 +110,25 @@ exports.plugin = {
             return reply.response(res.output.payload).code(statusCode);
           }
           // custom redirect https://github.com/dwyl/hapi-error/issues/5
-          if (config.statusCodes[statusCode]
-            && config.statusCodes[statusCode].redirect) {
+          var currentCodeConfig = config.statusCodes[statusCode];
+          if (currentCodeConfig && currentCodeConfig.redirect) {
             // if parameter is function, process it and redirect exactly as given (unless false or null)
-            if (typeof config.statusCodes[statusCode].redirect === 'function') {
-              let redirectPath = config.statusCodes[statusCode].redirect(request);
+            if (isFunction(currentCodeConfig.redirect)) {
+              let redirectPath = currentCodeConfig.redirect(request);
               if (redirectPath) {
                 return reply.redirect(redirectPath);
               }
-            } else if (config.statusCodes[statusCode].redirect) { // if parameter is string, append redirect query
-              return reply.redirect(config.statusCodes[statusCode].redirect + '?redirect=' + request.url.path);
+            } else if (currentCodeConfig.redirect) { // if parameter is string, append redirect query
+              var redirectString = request.url.pathname + request.url.search;
+              return reply.redirect(currentCodeConfig.redirect + '?redirect=' + redirectString);
```

Comment:
This appears to solve the issue of having an `undefined` value in the `redirect` URL param https://github.com/dwyl/hapi-error/issues/69 👍